lib/irc.js

```diff
@@ -790,6 +799,29 @@ Client.prototype.connect = function(retryCount, callback) {
         self.emit('netError', exception);
     });
 };
+
+Client.prototype.enableParseMiddleware = function(middleware) {
+    if (middleware && typeof middleware === 'function') {
+        this._parseMiddlewares.push(middleware);
+    }
+};
+
+Client.prototype.disableParseMiddleware = function(middleware) {
+    if (middlware && typeof middleware === 'function') {
+        this._parseMiddlewares = this.parseMiddlewares.filter(function(func) {
+            return func !== middleware;
```

Comment:
Okay, that makes good sense; consider my suggestion withdrawn.

On Sun, 11 Jan 2015 21:13:14 -0800
Justin Gallardo <notifications@github.com> wrote:

> > @@ -790,6 +799,29 @@ Client.prototype.connect =
> > function(retryCount, callback) { self.emit('netError', exception);
> >      });
> >  };
> > +
> > +Client.prototype.enableParseMiddleware = function(middleware) {
> > +    if (middleware && typeof middleware === 'function') {
> > +        this._parseMiddlewares.push(middleware);
> > +    }
> > +};
> > +
> > +Client.prototype.disableParseMiddleware = function(middleware) {
> > +    if (middlware && typeof middleware === 'function') {
> > +        this._parseMiddlewares =
> > this.parseMiddlewares.filter(function(func) {
> > +            return func !== middleware;
> 
> Preserving the order that the middlewares were added is important, so
> an array made sense to me.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/martynsmith/node-irc/pull/295/files#r22775449